File: examples/R/post_request.R
#!/usr/bin/env Rscript
suppressMessages(library(scrapystreaming))

save_data <- function(response) {
    f <- file("outputs/post.json")
    # serialize the extracted data and close the spider
    write(response$body, f)
    close(f);
    close_spider()
}

create_spider("post", character(0))
send_request("http://httpbin.org/post", save_data, method = "POST", body = "Post data")

run_spider()

File: examples/Python/post_request.py
#! /usr/bin/env python
from utils import *


def main():

    status = parse_json(stdin.readline())

    # we start checking if the channel is ready
    if status['status'] != 'ready':
        raise Exception("There is problem in the communication channel")

    write_line('''
        {
            "type": "spider",
            "name": "post",
            "start_urls": []
        }
        ''')

    write_line('''
            {
                "type": "request",
                "id": "post",
                "url": "http://httpbin.org/post",
                "method": "POST",
                "body": "Post data"
            }''')

    response = stdin.readline()
    data = parse_json(response)

    with open('outputs/post.json', 'w') as f:
        f.write(data['body'])
        f.flush()


if __name__ == '__main__':
    main()
